### PR TITLE
Add support for studio.json file for export import sites

### PR DESCRIPTION
**Pull request by sejas**
- Closes https://github.com/Automattic/dotcom-forge/issues/8362

## Proposed Changes

Adding Studio.json file into the export zip has been implemented on 

In this PR we:
- Parse studio.json file to read the new data
- Use the phpVersion from sutdio.json for new sites with correct import zips
- Create a useSiteDetails callback updateSiteState to update a site only on React state and not write the appData file.

## Testing Instructions

**Import existing site**
-  Run `STUDIO_IMPORT_EXPORT=true npm start`
- In the settings tab, change the PHP version to be something different, for example 7.4
- Go to Import/Export tab and export the entire site
- Add a new site
- Go to Import/Export tab
- Import  the backup file you exported previously
- Observe the existing site has the PHP version indicated in studio.json, in this example 7.4

https://github.com/user-attachments/assets/f0afa6b7-d4ec-4482-9dc0-519d861b337f

**Import a new site**
-  Run `STUDIO_IMPORT_EXPORT=true npm start`
- In the settings tab, change the PHP version to be something different, for example 7.4
- Go to Import/Export tab and export the entire site
- Click on Add site
- Select the backup file you exported previously
- Click Add site
- Observe the site created runs the PHP version indicated in studio.json, in this example 7.4

https://github.com/user-attachments/assets/55e57b8b-08d8-428e-8392-7725d656e8a5

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: src/hooks/use-add-site.ts
Comment (fluiddot) [inline src/hooks/use-add-site.ts]: @sejas I wonder if we could rely on the import logic to update the PHP version after the site is created, similarly to how it works when importing for an already-created site. This way we avoid the need to extract the metafile from the import before running the import process, which would simplify the logic. WDYT?

Comment (fluiddot) [reply src/hooks/use-add-site.ts]: I also noticed that this implies extracting the backup two times which might increase significantly the duration of the creation/import process.

Additionally, I observed that after adding a site from a big import file, there's a delay when the loading screen is shown. If we follow this approach, we'd need to present the loading screen during this process.


https://github.com/user-attachments/assets/4b5e3f34-5399-4e45-9103-983cf7ad93d8

File: src/hooks/use-import-export.tsx
Comment (fluiddot) [inline src/hooks/use-import-export.tsx]: The site export only produces the SQL file, so currently we are not exporting the PHP version in this case.

I presume the PHP version (as well as the WP version) are core dependencies that should be always synced when importing, but I wonder if the PHP version should be updated when only exporting the database 🤔.

File: src/lib/import-export/import/import-manager.ts
Comment (fluiddot) [inline src/lib/import-export/import/import-manager.ts]: If the backup file is big enough, we could consider reflecting the extraction of the metafile in the import progress bar as it will take some time. Hence, we could consider adding events to track the start and complete steps of the process.

**Review by fluiddot — COMMENTED** [review]
I went through the testing instructions and confirmed it works as expected. Great work @sejas. I hold the approval in case we'd like to address the issue I mentioned [here](https://github.com/Automattic/studio/pull/407#discussion_r1705181063) when importing big files.

File: src/hooks/use-add-site.ts
Comment (wojtekn) [reply src/hooks/use-add-site.ts]: It doesn't sound optimal to unpack the file two times during the import. I think we should fix it before we merge.

File: src/hooks/use-import-export.tsx
Comment (wojtekn) [reply src/hooks/use-import-export.tsx]: I would leave that as is - if users export SQL dump, they will be responsible for importing it to a correct PHP and WP version.

**Comment by sejas** [comment]
> @sejas I wonder if we could rely on the import logic to update the PHP version after the site is created, similarly to how it works when importing for an already-created site. This way we avoid the need to extract the metafile from the import before running the import process, which would simplify the logic. WDYT?

@fluiddot , thanks for reviewing it. Yeah,  I agree that avoiding extracting the backup twice will be a better approach. I took the approach of creating a site with the correct phpVersion from the beginning so the server starts with the correct version. 
I'll work to refactor it. We may need to wait until we merge the PR that starts the server after the import.

**Comment by sejas** [comment]
Trying to minimize conflicts, I've moved the export `studio.json` logic to https://github.com/Automattic/studio/pull/430
I'll work in the import logic reading `studio.json` after we merge the rest of open PRs that are refactoring heavily the `importer.ts`.

**Comment by sejas** [comment]
I've simplified the logic of this PR a lot, now it only unzips the backup once and it updates the site details after the site is created and imported. @fluiddot , let me know what you think.

**Comment by kozer** [comment]
Tested it and worked as expected! Nice work @sejas !!!

**Review by kozer — APPROVED** [review]

File: src/ipc-handlers.ts
Comment (fluiddot) [inline src/ipc-handlers.ts]: I wonder if we could let the caller (in this case the hook `useImportExport`) handle all the updates for the site after the import. This way we won't need to update it in two phases:
1. Here we update the site details in the site server and user data.
2. In `useImportExport` hook, with the returned data, we update the site details in the state of `useSiteDetails` hook.

I think this IPC handler should be only in charge of the import process and be agnostic to what needs to be updated on the site details.

WDYT?

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I followed the testing instructions and confirmed the PHP version defined in `studio.json` is applied when importing sites. I shared [this comment](https://github.com/Automattic/studio/pull/407#discussion_r1708893358) regarding a potential update of the changes but it's just an enhancement so it shouldn't block merging this PR.

File: src/ipc-handlers.ts
Comment (sejas) [reply src/ipc-handlers.ts]: Thank you for this suggestion. I think it's wise to re-use the same logic that is already implemented 👍 .
I've removed `updateSiteState` in favor of `updateSite`.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 ! Thanks @sejas for addressing my recent comments 💯 !

**Pull request merged**
